cmd/gke-gcloud-auth-plugin/cred/cred.go

```diff
@@ -231,9 +246,15 @@ func (p *plugin) getCachedGcloudAccessToken() (string, *metav1.Time, error) {
 	if err != nil {
 		return "", nil, fmt.Errorf("error retrieving starting config: %w", err)
 	}
+
 	// If current context is not the same as what the cached access token was
 	// generated for, then consider the current access token invalid.
-	if c.CurrentContext != startingConfig.CurrentContext {
+	currentContext := startingConfig.CurrentContext
+	if providerContext := p.tokenProvider.context(); providerContext != "" {
```

Comment:
Is this the only check that ensures cached tokens aren't returned to a different provider? If so, can we have a more reliable check than string matching for the provider-specificity? In particular, it seems like an empty-string provider context is valid, so if multiple providers fail to specify their own strings, then they might inadvertently share tokens.

Reply:
I thought a string field was a simple way to provide flexibility, considering that:
1) For cloud GKE, no distinction between tokens is necessary
2) For GDCE, we must verify token is created for same location and cluster
3) We could imagine other providers having different sets of necessary properties to be distinct